lib/core/src/irods_server_properties.cpp

```diff
@@ -22,11 +32,127 @@ namespace irods
     const std::string PROXY_USER_ZONE_KW( "proxy_user_zone" );
     const std::string PROXY_USER_PRIV_KW( "proxy_user_priv" );
     const std::string SERVER_CONFIG_FILE( "server_config.json" );
+    const std::string ENDPOINT_KW("IRODS_SERVER_CONFIGURATION_ENDPOINT");
+
+
+    namespace {
+		struct string_t
+        {
+			char*  ptr;
+			size_t len;	/* not counting terminating null char */
+		};
+
+		size_t curl_write_str(void *ptr, size_t size, size_t nmeb, void *stream)
+		{
+				size_t    new_len{};
+				string_t* string{};
+				void*     tmp_ptr{};
+
+				if (!stream) {
+						rodsLog (LOG_ERROR, "%s", "irodsCurl::write_string: NULL destination stream.");
+						return 0;
+				}
+
+				string = static_cast<string_t *>(stream);
+
+				new_len = string->len + size*nmeb;
+
+				/* Reallocate memory with space for new content */
+				/* Add an extra byte for terminating null char */
+				tmp_ptr = realloc(string->ptr, new_len + 1);
+				if (!tmp_ptr) {
+						rodsLog(LOG_ERROR, "%s", "irodsCurl::write_string: realloc failed.");
+						return -1;
```

Comment:
anything better than -1?

Reply:
looks like it should be 0, according to the curl documentation: https://curl.se/libcurl/c/CURLOPT_WRITEFUNCTION.html